shells/dev-shell/index.js

```diff
@@ -73,9 +73,12 @@ async function wrappedExecute() {
     arcPanel.attachArc(arc);
 
     const resolver = new RecipeResolver(arc);
-    const resolvedRecipe = await resolver.resolve(recipe);
+    const options = {errors: new Map()};
+    const resolvedRecipe = await resolver.resolve(recipe, options);
     if (!resolvedRecipe) {
-      arcPanel.showError('Error in RecipeResolver');
+      arcPanel.showError(`Could not normalize a context recipe:\n${
+        [...options.errors.entries()].join('\n')
+      }.\n${recipe.toString()}`);
```

Comment:
The error panel has a header and separate display for details, so this should just be `arcPanel.showError('Error resolving recipe', recipe.toString());`

Reply:
Thanks (done).